bugsnag-plugin-android-ndk/src/main/jni/bugsnag_ndk.c

```diff
@@ -196,7 +198,20 @@ Java_com_bugsnag_android_ndk_NativeBridge_deliverReportAtPath(
   }
   remove(event_path);
   (*env)->ReleaseStringUTFChars(env, _report_path, event_path);
-  pthread_mutex_unlock(&bsg_native_delivery_mutex);
+  goto exit;
+
+  exit:
+    pthread_mutex_unlock(&bsg_native_delivery_mutex);
+    if (jpayload != NULL) {
+      (*env)->ReleaseByteArrayElements(env, jpayload, (jbyte *)payload,
+                                       0); // <-- frees payload
+    }
+    if (jstage != NULL) {
+      (*env)->ReleaseByteArrayElements(
+          env, jstage, (jbyte *)event->app.release_stage, JNI_COMMIT);
+    }
+    (*env)->DeleteLocalRef(env, jpayload);
```

Comment:
will this succeed if `jpayload` is NULL?

Reply:
Yep. The call has no effect if the reference is NULL: https://docs.oracle.com/en/java/javase/11/docs/specs/jni/functions.html#deletelocalref